### PR TITLE
Fix sighting context post-processing

### PR DESCRIPTION
**Pull request by 0snap**
###  :notebook_with_decorative_cover: Description

Fix a bug with sighting context post-processing (both custom sinks and context transformation). 

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/threatbus](https://docs.tenzir.com/threatbus), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Enable both live- and retro-matching and test interactively by passing sighting context into fever alertify (use example config).

### CODE REVIEW

File: apps/vast/pyvast_threatbus/pyvast_threatbus.py [hunk 1]
```diff
@@ -566,53 +566,90 @@ async def report_sightings(
...
+        _, ioc_value = split_object_path_and_value(indicator.pattern)
+    else:
+        # try to find the indicator value instead
+        ioc_value = (
+            sighting.x_threatbus_indicator_value
+            if ThreatBusSTIX2Constants.X_THREATBUS_INDICATOR_VALUE.value
+            in sighting.object_properties()
+            else None
+        )
+    if not ioc_value:
```
Comment (tobim) [inline apps/vast/pyvast_threatbus/pyvast_threatbus.py:630]: I don't understand how this works. `ioc_value` has not been declared in this scope.

File: apps/vast/CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,9 @@ Every entry has a category for which we use the following visual abbreviations:
 
 ## Unreleased
 
+- 🐞 We fixed the post-processing of sighting context data for both retro- and
+  live-matched sightings.
```
Comment (dominiklohmann) [inline apps/vast/CHANGELOG.md:16]: Please explain more clearly what was broken.

File: apps/vast/pyvast_threatbus/pyvast_threatbus.py [hunk 3]
```diff
@@ -542,8 +543,8 @@ async def report_sightings(
     @param transform_cmd The command to use to pipe sightings to. Treated
         as template string: occurrences of '%ioc' in the cmd string get replaced
         with the matched IoC.
-    @param report_data If True, only report sighting.context.data instead of the
-        whole sighting.
+    @param report_data If True, only report context data of the sighting instead.data instead of the
```
Comment (dominiklohmann) [inline apps/vast/pyvast_threatbus/pyvast_threatbus.py:546]: The formatting looks a bit off here.

File: apps/vast/pyvast_threatbus/pyvast_threatbus.py (hunk 1 above)
Comment (0snap) [reply apps/vast/pyvast_threatbus/pyvast_threatbus.py:630]: Scoping in Python is different than in C++. e.g., `if` does not scope. Instead, variables declared inside an `if`, `while` etc have the scope of the enclosing function. See LEBG scope (google) or this [SO answer](https://stackoverflow.com/a/2829642/2525184)

Comment (tobim) [reply apps/vast/pyvast_threatbus/pyvast_threatbus.py:630]: TIL. :confused:

File: apps/vast/CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,11 @@ Every entry has a category for which we use the following visual abbreviations:
 
 ## Unreleased
 
+- 🐞 We fixed an unhandled exception in the post-processing of sighting context
+  data for both retro- and live-matched sightings. The bug was introduced with
+  the STIX-2 rewrite and effective rendered both the `transform_context` and
```
Comment (dominiklohmann) [inline apps/vast/CHANGELOG.md:17]:
```suggestion
  the STIX-2 rewrite and effectively rendered both the `transform_context` and
```

**Review by dominiklohmann — APPROVED** [review]
This took me a while to get running, but it seems to work. Not sure if I tested everything, but I managed to go use the previously defunct options.

There's one spelling mistake in the changelog entry, and I'd like to see a new story for creating an integration test for this feature so it's not as complicated to test. Afterwards feel free to merge.

**Pull request merged**